### PR TITLE
Adding className to carousel components

### PR DESCRIPTION
**Pull request by mkalish**
Addresses Issue #669 
- Adds support for carousel components to accept the className prompt
- Adds an example showing use of custom carousel tag and className

### CODE REVIEW

File: src/Carousel.js [hunk 1]
```diff
@@ -190,13 +191,15 @@ Carousel.propTypes = {
...
   slide: PropTypes.bool,
   cssModule: PropTypes.object,
+  className: PropTypes.string,
 };
 
 Carousel.defaultProps = {
   interval: 5000,
   pause: 'hover',
   keyboard: true,
   slide: true,
```
Comment (TheSharpieOne) [inline src/Carousel.js]: Don't think you meant to put this here

File: src/CarouselCaption.js [hunk 2]
```diff
@@ -4,11 +4,12 @@ import classNames from 'classnames';
...
 
 const CarouselCaption = (props) => {
-  const { captionHeader, captionText, cssModule } = props;
+  const { captionHeader, captionText, cssModule, className } = props;
   const classes = mapToCssModules(classNames(
+    className,
     'carousel-caption',
     'd-none',
-    'd-md-block'
+    'd-md-block',
```
Comment (TheSharpieOne) [inline src/CarouselCaption.js]: Idk if comma dangle in a function call is a thing

File: docs/lib/examples/CarouselCustomTag.js [hunk 3]
```diff
@@ -76,14 +76,13 @@ class Example extends Component {
   
       return (
         <div>
-          <style dangerouslySetInnerHTML={{__html: `
+          <style>{`
             .custom-tag { 
-                max-width: 100%px;
```
Comment (mkalish) [inline docs/lib/examples/CarouselCustomTag.js]: Good catch @TheSharpieOne

File: docs/lib/examples/CarouselCustomTag.js [hunk 4]
```diff
@@ -76,14 +76,13 @@ class Example extends Component {
   
       return (
         <div>
-          <style dangerouslySetInnerHTML={{__html: `
+          <style>{`
```
Comment (mkalish) [inline docs/lib/examples/CarouselCustomTag.js]: This is much better than what I had... I have a few other places to change now.

**Comment by jeremy-donson** [comment]
Please excuse the uninformed question.
Will this allow me to style a carousel image?

In particular, resizing it so that the image is responsive as well?

**Pull request merged**